authors/apps/articles/views.py

```diff
@@ -46,3 +47,65 @@ def delete(self, request, slug):
         """This method overwrites the default django for an error message"""
         super().delete(self, request, slug)
         return Response({"message": "Article Deleted Successfully"})
+
+class FavouriteGenericAPIView( APIView):
+    serializer_class = FavouriteSerializer
+    permission_classes = (IsAuthenticated,)
+    queryset = Favourite.objects.all()
+
+    def post(self, request, slug):
+        '''method to favourte by adding to db'''
+        article = None
+        try:
+            article = ArticlesModel.objects.get(slug=slug)
+        except:
+            data = {
+                    "errors":{
+                        "article": [
+                        "does not exist"
+                        ]
+                            }
+                     }
+            return Response(data=data, status=status.HTTP_400_BAD_REQUEST)
+        favourite = {}
+        favourite["user"] = request.user.id
+        favourite["article"] = article.pk
+        serializer = self.serializer_class(data=favourite)
+        serializer.is_valid(raise_exception=True)
+        serializer.save()
+        article_serializer = ArticlesSerializers(instance=article)
+        data = {"article":article_serializer.data}
+        data["article"]["favourited"]=True
+        data["message"] = "favourited"
+        return Response(data, status.HTTP_200_OK)
+
+    def delete(self, request, slug):
+        '''Method to unfavourite by deleting from the db '''
+        article = None
+        try:
+            article = ArticlesModel.objects.get(slug=slug)
+        except:
+            data = {
```

Comment:
Hey @caveinn, I have gone through the PR and I believe there is a better way you could have handled these errors, use renderers instead to handle the responses. However, your implementation is pretty good.